src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
+    end
+  end
+
+  def select!(&block : K, V -> U)
+    delete_if{|k, v| !yield k, v }
```

Comment:
`I like` and `I don't like` isn't a problem, but the opposite. Programming isn't a religion, that should have a Bible, that everyone should follow. Rules shouldn't be created as quickly as possible just to exist and make programmer harder to write own code. Take a look at C/C++ -- they still have several styles at the same time because it's obvious that there shouldn't be the single one. Crystal as any another language needs years and the style-guide should be written but the most experienced guys, not like bbatsov or his backyard friends.